reader_concurrency_semaphore.cc

```diff
@@ -947,6 +950,53 @@ reader_concurrency_semaphore::reader_concurrency_semaphore(int count, ssize_t me
     , _max_queue_length(max_queue_length)
     , _serialize_limit_multiplier(std::move(serialize_limit_multiplier))
     , _kill_limit_multiplier(std::move(kill_limit_multiplier))
+    , _metrics({
+        {"database", {
+            seastar::metrics::make_gauge("reads_memory_consumption", [this] { return consumed_resources().memory; },
+                        seastar::metrics::description("Holds the amount of memory consumed by current read operations. "),
+                        {class_label(_name)}),
+
+            seastar::metrics::make_gauge("queued_reads", _stats.waiters,
+                        seastar::metrics::description("Holds the number of currently queued read operations."),
+                        {class_label(_name)}),
+
+            seastar::metrics::make_gauge("paused_reads", _stats.inactive_reads,
+                        seastar::metrics::description("The number of currently active reads that are temporarily paused."),
+                        {class_label(_name)}),
+
+            seastar::metrics::make_counter("paused_reads_permit_based_evictions", _stats.permit_based_evictions,
+                        seastar::metrics::description("The number of paused reads evicted to free up permits."
+                                        " Permits are required for new reads to start, and the database will evict paused reads (if any)"
+                                        " to be able to admit new ones, if there is a shortage of permits."),
+                        {class_label(_name)}),
+
+            seastar::metrics::make_counter("reads_shed_due_to_overload", _stats.total_reads_shed_due_to_overload,
+                        seastar::metrics::description("The number of reads shed because the admission queue reached its max capacity."
+                                        " When the queue is full, excessive reads are shed to avoid overload."),
+                        {class_label(_name)}),
+
+            seastar::metrics::make_gauge("disk_reads", _stats.disk_reads,
+                        seastar::metrics::description("Holds the number of currently active disk read operations. "),
+                        {class_label(_name)}),
+
+            seastar::metrics::make_gauge("sstables_read", _stats.sstables_read,
+                        seastar::metrics::description("Holds the number of currently read sstables. "),
+                        {class_label(_name)}),
+
+            seastar::metrics::make_counter("total_reads", _stats.total_successful_reads,
+                        seastar::metrics::description("Counts the total number of successful user reads on this shard."),
+                        {class_label(_name)}),
+
+            seastar::metrics::make_counter("total_reads_failed", _stats.total_failed_reads,
+                        seastar::metrics::description("Counts the total number of failed user read operations. "
```

Comment:
```suggestion
                        seastar::metrics::description("Counts the total number of failed user read operations on this shard. "
```

Reply:
We'd have to add this clause to every metric. All our metrics are per shard, and there is an automatic label that specifies the shard.

(plus, this is just code movement)